### PR TITLE
Implement session quota restrictions

### LINKED ISSUE #86: Implement session quota restrictions
# Overview

The actions that can be taken through the API will need to have some quota restrictions put in place:
- distributedBlindedPepper: once per session
- requestSubsidisedAttestation: 10 per session* (not calls but attestations - the count is part of the request body) 
- submitMetaTransaction: 20 per session (calls)

> *requestSubsidisedAttestations - in order to actually track these correctly we would need to wait for the transaction to finalise in order to see if attestations where actually requested. But because this is a soft limit anyway (a new session can be created) we can start with a soft count as well for the MVP and just count them without ensuring that the transaction didn't revert.

If any of these limits are exceeded the API endpoint should throw a QuotaExceededError - this will be decoded by the client and Valora will initiate a new session.

### PR DESCRIPTION
**Pull request by Pedro-vk**
Closes #86

### CODE REVIEW

File: apps/onboarding/src/session/session.service.ts [hunk 1]
```diff
@@ -51,4 +66,15 @@ export class SessionService {
       return existingSession
     }
   }
+
+  async checkSessionQuota(session: Session, quota: SessionQuota, updateUsage: boolean = true) {
+    const usage = session.checkQuota(quota, updateUsage)
+    if (updateUsage) {
+      await this.update(session.id, session)
```
Comment (bowd) [inline apps/onboarding/src/session/session.service.ts:73]: Like I said before I would remove the update from the `session.checkQuota`. And handle the update here.

File: apps/onboarding/src/session/session.service.ts [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
 import { QueryDeepPartialEntity } from 'typeorm/query-builder/QueryPartialEntity'
 import { v4 as uuidv4 } from 'uuid'
-import { Session } from './session.entity'
+import { Session, SessionQuota } from './session.entity'
 import { SessionRepository } from './session.repository'
+import { quotaConfig } from '../config/quota.config'
+import { ApiError } from '../errors/api-error'
 
+class QuotaExceededError extends ApiError<'QuotaExceededError'> {
+  statusCode = 403
```
Comment (bowd) [inline apps/onboarding/src/session/session.service.ts:12]: Let's make it a [429](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429)

**Review by bowd — CHANGES_REQUESTED** [review]
Great work! I've added a few comments here and there.

**Comment by bowd** [comment]
@Pedro-vk I took this over in the weekend because it was blocking on adding another endpoint for the frontend work 🙏

File: apps/onboarding/src/errors/api-error.ts [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import { RootError } from '@celo/base/lib/result'
 
+export const apiErrorSymbol = Symbol('ApiError')
+
 export abstract class ApiError<TError, TMetadata = never> extends RootError<TError> {
-  _apiError: boolean = true
+  _apiError: Symbol = apiErrorSymbol
```
[the remark this answers is not in the document]
Comment (Pedro-vk) [reply apps/onboarding/src/errors/api-error.ts]: I think that your second example is the most common use. Could be a string, enum or other thing inside the symbol property. But yes 👌

File: apps/onboarding/src/session/session.entity.ts [hunk 4]
```diff
@@ -44,5 +51,13 @@ export class Session {
     isOpen(): boolean {
         return !this.expiredAt && !this.completedAt
     }
+
+    checkQuota(quota: SessionQuota, updateUsage: boolean = true) {
```
[the remark this answers is not in the document]
Comment (Pedro-vk) [reply apps/onboarding/src/session/session.entity.ts]: It is updated in the service, there is a good way to update a Entity from its own code?

**Comment by Pedro-vk** [comment]
> @Pedro-vk I took this over in the weekend because it was blocking on adding another endpoint for the frontend work 🙏 

Sure, feel free 👌

**Pull request merged**